normandy/recipes/models.py

```diff
@@ -122,18 +163,52 @@ def update_signature(self):
 
     @transaction.atomic
     def update(self, force=False, **data):
```

Comment:
the revision id is unique so you wouldn't match an older revision with the same content. the `latest_revision` is always the last revision. if it is not, something has gone terribly wrong.